### ISSUE TITLE
Explicit cast to int needed in Count() method in SimpleQuery.cs

### ISSUE THREAD

**Issue by taylorjg** [ThatRendle/Simple.Data#95]
I have been happily adding unit tests for my Informix ADO provider until I hit a problem with the following line of code:

```
        Assert.AreEqual(28, db.Customers.GetCount());
```

This fails with an exception:

Microsoft.CSharp.RuntimeBinder.RuntimeBinderException
"Cannot implicitly convert type 'decimal' to 'int'. An explicit conversion exists (are you missing a cast?)"

I tracked it down to this method in SimpleQuery.cs:

```
public int Count()
{
    return Select(new CountSpecialReference()).ToScalar();
}
```

Ultimately, the problem is that the underlying Informix ADO.NET Provider returns a decimal object from ExecuteScalar as the following little test program revealed:

```
    using (var cn = new IfxConnection(CONNECTION_STRING)) {
        cn.Open();
        var command = cn.CreateCommand();
        command.CommandType = CommandType.Text;
        command.CommandText = "select COUNT(*) from \"customer\"";
        object scalarResult = command.ExecuteScalar();
    }
```

Response from Mark:

"If you can raise an issue on Github, I'll add the explicit cast in. Cheers, Mark"

**Comment by ThatRendle** [comment]
Fixed in 0.11.2.

**Comment by flq** [comment]
Hmm, need to add a test for this in the Oracle provider as it ultimately suffers from the same fate of having all numbers returned as decimal...
